pkg/kubelet/network/cni/cni.go

```diff
@@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
 	host           network.Host
 	execer         utilexec.Interface
 	nsenterPath    string
+	mu             sync.Mutex
+	podMutexes     map[string]*sync.Mutex
```

Comment:
Your explanation REALLY makes it sound like it belongs up-stack.  Looking at kubelet really quickly, it doesn't look any more complicated to do it there (though you'll have to do it in both docker and rkt code).

@yujuhong